src/App.tsx

```diff
@@ -10,32 +12,28 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTab, setSelectedTab] = useState(tabs[0]);
+
+  const onTabSelected = (tab: Tab) => {
+    setSelectedTab(tab);
+  };
```

Comment:
Let's use `useCallback` for rendering optimization, cause you are passing the callback to another component